core/src/test/java/org/apache/accumulo/core/conf/cluster/ClusterConfigParserTest.java

```diff
@@ -242,4 +244,23 @@ public void testShellOutputWithOptionalComponents() throws Exception {
     assertEquals(expected, actual);
   }
 
+  @Test
+  public void testFileWithUnknownSections() throws Exception {
+    URL configFile = ClusterConfigParserTest.class
+        .getResource("/org/apache/accumulo/core/conf/cluster/bad-cluster.yaml");
+    assertNotNull(configFile);
+
+    Map<String,String> contents =
+        ClusterConfigParser.parseConfiguration(new File(configFile.toURI()).getAbsolutePath());
+
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    PrintStream ps = new PrintStream(baos);
+
+    var exception = assertThrows(IllegalArgumentException.class,
+        () -> ClusterConfigParser.outputShellVariables(contents, System.out));
+
+    assertTrue(exception.getMessage().contains("vserver"));
+
+    ps.close();
```

Comment:
try-with-resources is shorter:

```suggestion
    try (var baos = new ByteArrayOutputStream(); var ps = new PrintStream(baos)) {
      var exception = assertThrows(IllegalArgumentException.class,
          () -> ClusterConfigParser.outputShellVariables(contents, System.out));
      assertTrue(exception.getMessage().contains("vserver"));
    }
```